### PR TITLE
Validate open() mode

### LINKED ISSUE #5895: `open()` does not report invalid modes
A user did `open("filename", "b")`. No error is reported by the `open()` call, but later, when the file handle was used, bad things can happen, e.g.:
```py
>>> f = open("bad.py", "b")
>>> m = audiomp3.MP3Decoder(f)
Traceback (most recent call last):
```
(that is the whole traceback!)

The CPython error for the above:
```
ValueError: Must have exactly one of create/read/write/append mode and at most one plus
```
For a completely invalid mode (bad character):
```
ValueError: invalid mode: 'z'
```

We don't have to be as exact as CPython, but we should do better.

### PR DESCRIPTION
**Pull request by dhalbert**
- Fixes #5895 

Checks validity of the `mode` argument to `open()`. Prevents passing bad modes further down.

This was a `// TODO` from MicroPython.

Tested in the REPL with good and bad mode strings.

### CODE REVIEW

File: extmod/vfs_fat_file.c [hunk 1]
```diff
@@ -139,39 +139,59 @@ STATIC const mp_arg_t file_open_args[] = {
...
                 type = &mp_type_vfs_fat_fileio;
                 break;
             #endif
             case 't':
+                bt_count++;
                 type = &mp_type_vfs_fat_textio;
                 break;
+            default:
+                bad_mode = true;
+                mp_raise_ValueError(translate("Invalid mode"));
```
Comment (tannewt) [inline extmod/vfs_fat_file.c:180]:
```suggestion
```

Rely on the `if` below for the error. That way it's raised in one spot.

**Review by tannewt — CHANGES_REQUESTED** [review]
Thank you for this improvement! One request

File: extmod/vfs_fat_file.c (hunk 1 above)
Comment (dhalbert) [reply extmod/vfs_fat_file.c:180]: My typo! That's what I mean to do. Fixed.

**Review by tannewt — APPROVED** [review]
Thank you!

**Comment by dhalbert** [comment]
@tannewt ready for review again: I had to shrink one build to fit.

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**